compiler/src/iree/compiler/Dialect/Stream/Transforms/test/refine_usage.mlir

```diff
@@ -210,3 +210,75 @@ func.func @escapingAlloca() -> !hal.buffer_view {
   %2 = stream.tensor.export %1 : tensor<f32> in !stream.resource<external>{%c123} -> !hal.buffer_view
   return %2 : !hal.buffer_view
 }
+
+// -----
+
+// CHECK-LABEL: @testIf
+func.func @testIf(%arg0: i1, %arg1: !hal.buffer_view, %arg2: !hal.buffer_view) -> !hal.buffer_view attributes {iree.abi.stub} {
+  %c4 = arith.constant 4 : index
+  %c0 = arith.constant 0 : index
+  %c268435488_i32 = arith.constant 268435488 : i32
+  %c1_i32 = arith.constant 1 : i32
+  %c1 = arith.constant 1 : index
+  hal.buffer_view.assert<%arg1 : !hal.buffer_view> message("input 1") shape([%c1]) type(%c268435488_i32) encoding(%c1_i32)
+  %0 = stream.tensor.import %arg1 : !hal.buffer_view -> tensor<1xi32> in !stream.resource<external>{%c4}
+  %1 = stream.async.transfer %0 : !stream.resource<external>{%c4} -> !stream.resource<*>{%c4}
+  hal.buffer_view.assert<%arg2 : !hal.buffer_view> message("input 2") shape([%c1]) type(%c268435488_i32) encoding(%c1_i32)
+  %2 = stream.tensor.import %arg2 : !hal.buffer_view -> tensor<1xi32> in !stream.resource<external>{%c4}
```

Comment:
you can simplify this and remove the hal ops - just pass in the `!stream.resource<external>`s (we generally don't want to include other dialect ops, and we don't need this portion to test the changes)

(here and below)

Reply:
Makes sense. The test was originally just from a dumped simple example, so having a more readable test should help.